### PR TITLE
PC-1349: add "property type" to EPC page

### PR DESCRIPTION
**Pull request by Glenn-Clarke**
[Link to Jira ticket](https://beisdigital.atlassian.net/browse/PC-1373)

# Description

- Property type from the EPC will now be displayed in the blue box
- It will translate to Welsh when required
- If no translated property type is listed, an error will be logged, and no property type will be shown.

# Checklist

- [x] I have made any necessary updates to the documentation
- [x] I have checked there are no unnecessary IDE warnings in this PR
- [x] I have checked there are no unintentional line ending changes
- [x] I have added tests where applicable
- [x] If I have made any changes to the code, I have used the IDE auto-formatter on it
- [x] If I have made any changes to Python files (even if not changing any content strings), I have run `make extract-translations`
- [x] If I have made any changes to website flow, I have updated the [Flow Miro Board](https://miro.com/app/board/uXjVK4AgQ3E=/)
- [x] If I have made any changes to website flow, I have checked forward and back behaviour is still consistent

# Screenshots

image

image

### CODE REVIEW

File: help_to_heat/frontdoor/views.py [hunk 1]
```diff
@@ -835,6 +843,14 @@ def build_extra_context(self, request, session_id, page_name, data, is_change_pa
         epc_band = epc.get("current-energy-rating")
         epc_date = epc.get("lodgement-date")
 
+        epc_property_type = epc.get("property-type").upper()
+
+        if epc_property_type in property_types:
+            property_type = property_types[epc_property_type]
+        else:
+            logger.error(f"Unrecognised Property Type: {epc_property_type}")
+            property_type = ""
```
Comment (samyou-softwire) [inline help_to_heat/frontdoor/views.py:852]: could there be a more ergonomic fallback value here? None?

Comment (Glenn-Clarke) [reply help_to_heat/frontdoor/views.py]: I think None is easier to read, yeah. I've made the change for the Date field as well, so both fallback values are consistent.

**Review by samyou-softwire — APPROVED** [review]

File: help_to_heat/frontdoor/views.py [hunk 2]
```diff
@@ -835,13 +843,21 @@ def build_extra_context(self, request, session_id, page_name, data, is_change_pa
         epc_band = epc.get("current-energy-rating")
         epc_date = epc.get("lodgement-date")
 
+        epc_property_type = epc.get("property-type").upper()
+
+        if epc_property_type in property_types:
+            property_type = property_types[epc_property_type]
+        else:
+            logger.error(f"Unrecognised Property Type: {epc_property_type}")
+            property_type = None
```
[the remark this answers is not in the document]
Comment (Glenn-Clarke) [reply help_to_heat/frontdoor/views.py:852]: Yes, we can. Is that fine to do (on the assumption we fix it quickly if the errors come through) when it comes to Welsh Language compliance?

[the remark this answers is not in the document]
Comment (jdgage) [reply help_to_heat/frontdoor/views.py]: Fair question, and not one I can confidently answer, so would have to ask DESNZ. I'd be surprised if we couldn't do this though.

[the remark this answers is not in the document]
Comment (Glenn-Clarke) [reply help_to_heat/frontdoor/views.py]: I've left a note on the ticket, and I'll raise it at standup tomorrow morning 😃

[the remark this answers is not in the document]
Comment (Glenn-Clarke) [reply help_to_heat/frontdoor/views.py]: If we can make these flag more often than currently (i.e. every time this happens?) so we can act on it (relatively) immediately, we can show this in English for that short period. 
If we can't, we should still log it but hide it.

[the remark this answers is not in the document]
Comment (jdgage) [reply help_to_heat/frontdoor/views.py]: Ok, we're not going to turn on alerts for every error (the cost of investigating each one would be silly). So it sounds like we should display nothing. Feels like cutting our nose off to spite our face, but there you go.

Do we have the option of only blanking it out if the user's looking at the welsh page? What currently happens if there's no translation for a string?

[the remark this answers is not in the document]
Comment (samyou-softwire) [reply help_to_heat/frontdoor/views.py]: on the latter, if there's no translation then the translation function returns the original string, though this is a little hard to disambiguate as this is also true if the language is set to english

on the former, seems easiest to check `django.utils.translation.get_language()`

File: tests/test_frontdoor.py [hunk 3]
```diff
@@ -2010,6 +2010,65 @@ def test_on_epc_select_page_manual_link_is_shown_if_no_addresses_found():
...
+    assert page.has_one("p:contains('Property type')")
+    assert page.has_one("p:contains('Maisonette')")
+
+    assert page.has_one("p:contains('EPC rating')")
+    assert page.has_one("p:contains('G')")
+
+    assert page.has_one("p:contains('Date of issue')")
+    assert page.has_one("p:contains('23 July 2010')")
+
+
```
Comment (samyou-softwire) [inline tests/test_frontdoor.py:2071]: would like to have added more tests for the english & welsh discrepancy, but testino struggles with setting a language. had to abandon those tests for now, though kept some of the refactors to test EPC loading

**Review by jdgage — APPROVED** [review]

**Pull request merged**